### PR TITLE
Implement `debugError` command

### LINKED ISSUE #640: Implement `debugError` command
We have a stub for the special `debugError` command. We should implement it for testing FerretDB error handling.

Its command handler should accept a single string parameter that would change its behavior:
* for `ok`, it should return a successful response;
* for a numerical value like `2` (still as string), it should return an error with that code;
* for `panic` it should panic;
* for any other value, it should return `errors.New(that_value)`.

The integration test should be in the dance repo (diff test) and cover the above cases plus error codes 1 (internal error) and 0 (that seems to be possible for `getParameter`).

We don't document that difference from MongoDB.

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #640.

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/common/msg_debugerror.go [hunk 1]
```diff
@@ -34,10 +37,29 @@ func MsgDebugError(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
 		return nil, err
 	}
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/640
+	// check if parameter is an error code
+	if n, err := strconv.Atoi(expected); err != nil {
```
Comment (w84thesun) [inline internal/handlers/common/msg_debugerror.go:41]: Shouldn't it be `err == nil`?

Comment (noisersup) [reply internal/handlers/common/msg_debugerror.go:41]: It should... nice catch!

**Review by w84thesun — APPROVED** [review]

**Comment by noisersup** [comment]
I have set donotmerge label because of the lack of dance tests

File: internal/handlers/common/msg_debugerror.go [hunk 2]
```diff
@@ -34,10 +37,29 @@ func MsgDebugError(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
 		return nil, err
 	}
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/640
+	// check if parameter is an error code
+	if n, err := strconv.Atoi(expected); err == nil {
+		errCode := ErrorCode(n)
```
Comment (rumyantseva) [inline internal/handlers/common/msg_debugerror.go:42]: What if `n` is something that is not present in the list of `ErrorCode`s? 
For example, if `n == 123456`, how would this work?

**Review by rumyantseva — COMMENTED** [review]
Asked a question.

File: internal/handlers/common/msg_debugerror.go (hunk 2 above)
Comment (AlekSi) [reply internal/handlers/common/msg_debugerror.go:42]: Good thing to cover with a dance test

Comment (noisersup) [reply internal/handlers/common/msg_debugerror.go:42]: @rumyantseva The ErrorCode.String() method will return it as `"ErrorCode(123456)"`.

**Review by AlekSi — APPROVED** [review]

**Review by chilagrow — APPROVED** [review]

**Comment by noisersup** [comment]
Dance tests are failing due to issues that are already on the main branch. When they are run individually, they pass. As agreed on the call - let's merge this PR and fix dance tests.

**Pull request merged**